tests/run-make/metadata-only-crate-no-ice/rmake.rs

```diff
@@ -0,0 +1,19 @@
+// In a dependency hierarchy, metadata-only crates could cause an Internal
+// Compiler Error (ICE) due to a compiler bug - not correctly fetching sources for
+// metadata-only crates. This test is a minimal reproduction of a program that triggered
+// this bug, and checks that no ICE occurs.
+// See https://github.com/rust-lang/rust/issues/40535
+
+use run_make_support::rustc;
+
+fn main() {
+    rustc().input("baz.rs").emit("metadata").run();
+    rustc().input("bar.rs").emit("metadata").extern_("baz", "libbaz.rmeta").run();
+    // There should be no internal compiler error message.
+    rustc()
+        .input("foo.rs")
+        .emit("metadata")
+        .extern_("bar", "libbaz.rmeta")
+        .run()
+        .assert_stderr_not_contains("unexpectedly panicked");
```

Comment:
Suggestion: this could just `assert_exit_code(1)` (ICE is 101).

Reply:
I think you mean `0`, as this is supposed to succeed, but yes.

Reply:
Actually, it failing at all is bad. I could really just remove the last line.